lib/Runtime/Library/JavascriptPromise.cpp

```diff
@@ -815,9 +782,18 @@ namespace Js
         ARGUMENTS(args, callInfo);
         Assert(!(callInfo.Flags & CallFlags_New));
 
-        Assert(args.Info.Count > 1);
+        Var arg;
+
+        if (args.Info.Count > 1 && args[1] != nullptr)
+        {
+            arg = args[1];
+        }
+        else
+        {
+            arg = function->GetScriptContext()->GetLibrary()->GetUndefined();
```

Comment:
Spec says the Thrower function should return a completion with the value equal to the argument. So, I think it's correct for this function to be called with no argument and return undefined. I'll add assert that the arg is not nullptr here.